### PR TITLE
Fix billing portal for users without stripeCustomerId

### LINKED ISSUE #1932: Billing portal is broken for users without metadata.stripeCustomerId
### Issue Summary

There are many users that don't have a stripeCustomerId on their `users.metadata` column. We should add a step in our logic so this happens:

1. If user doesn't have a `stripeCustomerId`.
2. Look him up by email and if found save his `stripeCustomerId`.
3. If not found, create a new customer with his email and save the new `stripeCustomerId`.

This should happen on upgrade, and billing portal pages.

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

When _stripeCustomerId_ isn't found on user.metada we call stripe api to create and save customerId on prisma user model.

Fixes #1932

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [X] Navigate to _/settings/billing_ page as a user without stripeCustomerId value, then click on "Go to the billing portal" button and it should navigate correctly to **billing.stripe.com** site.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code and corrected any misspellings
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by baileypumfleet** [comment]
@zomars Surely this's wouldn't work for us, as the customers without an ID already exist, so we would end up duplicating their Stripe accounts?

**Comment by zomars** [comment]
> @zomars Surely this's wouldn't work for us, as the customers without an ID already exist, so we would end up duplicating their Stripe accounts?

Not really. Right now we have users that don't exist as customers Stripe. Meaning that they don't have an account there. So this would check for that, create them and save them gradually.

**Review by zomars — APPROVED** [review]
Code looks simple and good, we would need to check if this works both for upgrading and the billing portal for users that don't have a customerId yet.

File: apps/web/ee/lib/stripe/customer.ts [hunk 1]
```diff
@@ -46,6 +46,21 @@ export async function getStripeCustomerId(user: UserType): Promise<string | null
     });
```
Comment (zomars) [inline apps/web/ee/lib/stripe/customer.ts:46]: There's a missing edge-case here. If we don't have the stripeCustomerId in our DB but it exists in Stripe, we need to save it in our DB as well.

**Review by zomars — CHANGES_REQUESTED** [review]

File: apps/web/ee/lib/stripe/customer.ts (hunk 1 above)
Comment (alannnc) [reply apps/web/ee/lib/stripe/customer.ts:46]: Just moved prisma user update to run for those 2 scenarios(Get from stripe and not on our db and newly created stripe.Customer).

**Review by zomars — CHANGES_REQUESTED** [review]

**Review by zomars — APPROVED** [review]

**Pull request merged**